pkg/proxy/client/client_conn.go

```diff
@@ -69,22 +70,22 @@ func (cc *ClientConnection) connectBackend(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if err = cc.connMgr.Connect(ctx, addr, cc.pkt, cc.frontendTLSConfig, cc.backendTLSConfig); err != nil {
+	if err = cc.connMgr.Connect(ctx, addr, cc.pkt, cc.frontendTLSConfig, cc.backendTLSConfig, cc.proxyProtocol); err != nil {
 		return err
 	}
 	return nil
 }
 
 func (cc *ClientConnection) Run(ctx context.Context) {
 	if err := cc.connectBackend(ctx); err != nil {
-		cc.logger.Info("new connection fails", zap.String("remoteAddr", cc.Addr()), zap.Error(err))
+		cc.logger.Info("new connection fails", zap.Error(err))
 		return
 	}
 
 	if err := cc.processMsg(ctx); err != nil {
 		clientErr := errors.Is(err, ErrClientConn)
 		if !(clientErr && errors.Is(err, io.EOF)) {
-			cc.logger.Info("process message fails", zap.String("remoteAddr", cc.Addr()), zap.Error(err), zap.Bool("clientErr", clientErr), zap.Bool("serverErr", !clientErr))
+			cc.logger.Info("process message fails", zap.Error(err), zap.Bool("clientErr", clientErr), zap.Bool("serverErr", !clientErr))
```

Comment:
You can assign the address in the proxy protocol to the ClientConnection so that it can log client addresses.
Or you can use `logger.With` to contain the real client address.

Reply:
I have done that, check `pkg/proxy/proxy.go`.